fastlane.gemspec

```diff
@@ -92,6 +92,13 @@ Gem::Specification.new do |spec|
   spec.add_dependency('simctl', '~> 1.6.3') # Used for querying and interacting with iOS simulators
   spec.add_dependency('jwt', '~> 2.1.0') # Used for generating authentication tokens for AppStore connect api
 
+  # need to lock 0.11 and under when using less than Ruby 2.4 to prevent install issues when using 'gem install'
+  # 'gem install' does not respect Ruby versions when and would try installing 0.12 on Ruby 2.3 or less
+  # https://github.com/fastlane/fastlane/pull/15483
+  if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.4')
+    spec.add_dependency('signet', '<= 0.11') # Because yeah
```

Comment:
See my other comment, I think we could add a log message here, something like 

```
WARN: Locking to a potentially insecure version of signet because you are using a version of ruby which is marked as End-Of-Life. Please upgrade your ruby installation to 2.4 or later
```

I don’t remember whether gem shows that during installation though, it might suppress output. Something worth to test though

Reply:
I will test it out 👌

Reply:
@milch I have to do an `STDERR.puts` for it to show (which I'm pretty sure is what the other warnings use that rubygems outputs

Reply:
I think `STDERR` is definitely appropriate for this 👌

Reply:
@milch Perfect! New commit added 😊

Reply:
Cool! 

As a side note, thinking about WWFD (what would @KrauseFx do), it’d probably be to figure out how the user installed ruby and then show them the commands they need to run to do the upgrade. I maybe wouldn’t do that right away ‘cause people may not need it, but may be something to consider if users are confused about what to do and start sending in issues about it